app/services/ItemsCachingService.scala

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+import com.google.inject.Inject
+import config.AppConfig
+import controllers.util.CacheIdGenerator.itemsId
+import forms.supplementary.ItemType.IdentificationTypeCodes._
+import forms.supplementary.{CommodityMeasure, ItemType, PackageInformation}
+import javax.inject.Singleton
+import models.DeclarationFormats._
+import models.declaration.supplementary.{AdditionalInformationData, DocumentsProducedData, ProcedureCodesData}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.cache.client.CacheMap
+import uk.gov.hmrc.wco.dec._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ItemsCachingService @Inject()(cacheService: CustomsCacheService)(appConfig: AppConfig) {
```

Comment:
I would advise to clean the code in this entire file. It is hard to read and to reason what is going on here, for e.g. splitting methods into smaller ones could be the first step.

Reply:
this is the easier version, the issue is with the way the screens are designed and how eco-dec model is and that is complicated I spent quite a time on this and handling this is not going to be easy and more over the designs are going to change in future and so as this class.